docs/design-doc-provider-identity-based-auth.md

```diff
@@ -0,0 +1,571 @@
+  account trust policy exists between the EKS cluster's OIDC provider and a
+  target IAM role in a different account (than the account owning the EKS
+  cluster and the OIDC provider), then it's possible to switch to an IAM role in
+  that target AWS account.
+- Privileges on the managed external resources need to be defined on the target
+  IAM roles of the STS `Assume*` operations. And as mentioned, fine-grained
+  access policies can be defined on these target roles which are configurable
+  with the `ProviderConfig` API.
+- When combined with the already available single-cluster multi-tenancy
+  techniques discussed above, this allows `provider-aws` users to isolate their
+  tenant identities and the privileges required for those identities.
+
+From the relevant discussions on `provider-aws` surveyed for this writing, this
+level of tenant isolation has mostly been sufficient for `provider-aws` users.
+But as discussed above, a deeper isolation is still possible. Especially in the
+currently feasible `provider-aws` authentication scheme, the initial
+`AssumeRoleWithWebIdentity` target IAM role is still shared by the tenants
+although it does not require privileges on the managed external resources. But
+due to vulnerabilities discussed in the [Limitations of Naming Convention-based
+or Admission Controller-based Approaches] section above, it could still be
+possible for a tenant to assume an IAM role with more privileges than it needs,
+starting with the shared `AssumeRoleWithWebIdentity` target IAM role. A deeper
+isolation between tenants would be possible if it were possible to have a
+Kubernetes service account and an associated (initial) non-shared IAM role
+assigned to each tenant.
+
+As of this writing, `provider-jet-aws` supports IRSA authentication with support
+for role chaining via the STS `AssumeRole` API operation. Similar to
+`provider-aws`, only chains of length `1` are allowed. Also, `provider-jet-aws`
+does not currently support specifying the target `AssumeRoleWithWebIdentity` IAM
+role via the `ProviderConfig` API. And unlike `provider-aws`, `provider-jet-aws`
+does not support specifying external IDs, session tags or transitive tag keys
+for the `AssumeRole` operation, or specifying session names for the
+`AssumeRoleWithWebIdentity` operation.
+
+**Azure**: Azure has the notion of system-assigned or user-assigned [managed
+identities][azure-msi], which allow authentication to any resource that supports
+Azure AD authentication. Some Azure services, such as EKS, allow a managed
+identity to be enabled directly on a service's instance (system-assigned). Or a
+user-assigned managed identity can be provisioned and assigned to the service
+instance. Similar to AWS IRSA, Azure has also introduced [Azure AD workload
+identities][azure-wi], which work in a similar way to IRSA:
+
+|     |
+| :-: |
+| <img src="images/azure-wi.png" alt="drawing" /> |
+| Azure AD Workload Identities (reproduced from [[1]]) |
```

Comment:
When I view the document, the image is displayed as corrupt/broken. I'm not sure what's causing this, could you please make sure this is displayed properly?


![Screenshot 2023-07-05 at 13 33 31](https://github.com/upbound/upjet/assets/103541666/ff240ded-79f3-4602-96af-08b8e84d41c3)

Reply:
Resolved